File: ly_thuyet/buoi1/nhi_phan_n_bit1.py
t=int(input())

for v in range(t):
    n, k=input().split()
    n, k=int(n), int(k)
    res, last=[], []
    for i in range(n):
        if(i<k):
            res.insert(0,'1')
        else:
            res.insert(0,'0')
        last.append('0')
    print(''.join(res))
    while res!=last:
        dem=0
        for i in range(n-1,-1,-1):
            if res[i]=='1':
                res[i]='0'
                continue
            if res[i]=='0':
                res[i]='1'
                break
        for i in range(n):
            if res[i]=='1':
                dem+=1
            if dem>k:
                break
        if dem==k:
            print(''.join(res))

File: thuchanh1/nhi_phan_n_bit1.cpp
#include<iostream>

using namespace std;

int main(){
    int t;
    cin>>t;
    while(t--){
        int n, k;
        cin>>n>>k;
        string res="", last="";
        for(int i=0;i<n;i++){
            if(i<k)
                res+='1';
            else
                res='0'+res;
            last+='1';
        }
        cout<<res<<endl;
        while(res!=last){
            int dem=0;
            for(int i=n-1;i>=0;i--){
                if(res[i]=='1'){
                    res[i]='0';
                    continue;
                }
                if(res[i]=='0'){
                    res[i]='1';
                    break;
                }
            }
            for(int i=0;i<n;i++){
                if(res[i]=='1')
                    dem++;
                if(dem>k)
                    break;
            }
            if(dem==k)
                cout<<res<<endl;
        }
    }
    return 0;
}